src/clients/ProfitClient.ts

```
@@ -460,11 +464,14 @@ export class ProfitClient {
     // Generate list of tokens to retrieve. Map by symbol because tokens like
     // ETH/WETH refer to the same mainnet contract address.
     const tokens: { [_symbol: string]: string } = Object.fromEntries(
-      this.hubPoolClient.getL1Tokens().map(({ symbol }) => {
-        const { addresses } = TOKEN_SYMBOLS_MAP[symbol];
-        const address = addresses[1];
-        return [symbol, address];
-      })
+      this.hubPoolClient
+        .getL1Tokens()
+        .filter(({ symbol }) => isDefined(TOKEN_SYMBOLS_MAP[symbol]))
```

Should we throw an error if a symbol is missing rather than silently filter it out?

---

Pre this change it would implicitly throw by trying to dereference undefined. I think it's preferable to filter because it permits the relayer to run even when a deposit defines a token that's not supported (this is especially annoying on testnet). We could choose to log in the event that a token is not resolvable; that might ease debugging in future.

---

We now emit a `warn` if we find any tokens that we expect to be in the hubpool but don't have a corresponding TOKEN_SYMBOLS_MAP entry.

I chose `warn` because any triggering of this warning should be sent on slack & should be corrected quickly.